src/common/GraphicalInformationSet.cpp

```diff
@@ -497,7 +497,11 @@ void GraphicalInformationSet::setDiscreteColorMap(AbstractObject* targetObject,
 
 	colorInformation->UseReverseMapping = useReverseMapping;
 	colorInformation->UseLogarithmicMapping = useLogarithmicMapping;
-	colorInformation->ValueVectorIndex = valueVectorIndex;
+
+	if (colorInformation->ValueVectorIndex == nullptr) {
+		colorInformation->ValueVectorIndex = soap_new_LONG64(gsoapProxy2_2->soap, 1);
```

Comment:
I think the second parameter of soap_new... is not needed (even if we use it often)

Reply:
Done

Reply:
Done.